compiler/luci/import/src/CircleReader.cpp

```diff
@@ -257,33 +276,46 @@ std::unique_ptr<SparsityParam> luci_sparsityparam(const circle::SparsityParamete
   return sparsityparam;
 }
 
-void copy_tensor_attributes(const circle::TensorT &tensor, CircleNode *node)
+std::unique_ptr<SparsityParam> luci_sparsityparam(const circle::SparsityParameters *sparparam)
 {
+  // create temporary unpacked API object
+  assert(sparparam != nullptr);
+  circle::SparsityParametersT sparsity;
+  sparparam->UnPackTo(&sparsity);
+
+  return luci_sparsityparam(&sparsity);
+}
+
+void copy_tensor_attributes(const circle::Tensor *tensor, CircleNode *node)
+{
+  assert(tensor != nullptr);
+
   node->name(tensor_name(tensor));
-  node->dtype(luci_datatype(tensor.type));
+  node->dtype(luci_datatype(tensor->type()));
 
-  assert(tensor.shape_signature.size() == 0 ||
-         tensor.shape_signature.size() == tensor.shape.size());
+  auto const tensor_shape_signature = wrap(tensor->shape_signature());
+  auto const tensor_shape = wrap(tensor->shape());
+  assert(tensor_shape_signature.is_null() || tensor_shape_signature.size() == tensor_shape.size());
 
-  std::vector<int32_t> dims = tensor.shape; // in NHWC
+  auto const dims = tensor_shape; // in NHWC
   node->rank(dims.size());
   for (uint32_t r = 0; r < dims.size(); ++r)
   {
-    if (tensor.shape_signature.size() > 0 && tensor.shape_signature.at(r) == -1)
+    if (not tensor_shape_signature.is_null() && tensor_shape_signature.at(r) == -1)
       node->dim(r).unset();
     else
       node->dim(r).set(dims[r]);
   }
 
-  const auto *quantization = tensor.quantization.get();
+  auto const quantization = tensor->quantization();
```

Comment:
`const auto *quantization` was to explicitly express it's a pointer and still `tensor->quantization();` seems too.
if this is so, please remove these kind of changes.

Reply:
Done.